presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergResourceFactory.java

```diff
@@ -99,25 +101,7 @@ public SupportsNamespaces getNamespaces(ConnectorSession session)
     private String getCatalogCacheKey(ConnectorSession session)
     {
         StringBuilder sb = new StringBuilder();
-        ConnectorIdentity identity = session.getIdentity();
-        sb.append("User:");
-        sb.append(identity.getUser());
-        if (identity.getPrincipal().isPresent()) {
-            sb.append(",Principle:");
-            sb.append(identity.getPrincipal());
-        }
-        if (identity.getRole().isPresent()) {
-            sb.append(",Role:");
-            sb.append(identity.getRole());
-        }
-        if (identity.getExtraCredentials() != null) {
-            identity.getExtraCredentials().forEach((key, value) -> {
-                sb.append(",");
-                sb.append(key);
-                sb.append(":");
-                sb.append(value);
-            });
-        }
+        sb.append(catalogName);
```

Comment:
`ref hash` is not exposed as Iceberg catalog property but it's not a hidden field from session config - https://github.com/prestodb/presto/blob/master/presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergSessionProperties.java#L277 

So the user can still make changes to it though as per Nessie doc, it's an optional field.  If we take it out from the cache key then it will load the cache value with the default `ref hash` i.e `null` so we don't have `ref hash` then should we also remove it from session properties in the connector?

Reply:
Is there a case that the nessie reference name itself is not unique and we have to use the reference hash to differentiate them? Or say, is it feasible to expect that reference names are unique in practical use cases?
Using a hash in a key for hashing sounds a bit uncommon to me...

Reply:
Not really sure if we really have to expose `ref.hash` property from Presto side if we have reference names there. But if we think we don't really need `ref.hash` then I think we should take it out as option from Presto session properties as well?

Reply:
@ChunxuTang The ref hash is the hash of the reference name. If the name is not unique, the hash won't be either. Also the hash is optional according to https://projectnessie.org/tools/client_config/. 

I actually don't understand why the "nessie_reference_hash" Iceberg session property was added at the first place. It does not have a corresponding config property in NessieConfig, and it is not settable and is defaulted to null.  There is no hashing applied to the reference name to set the "nessie_reference_hash" property, nor there was any verification it's a valid hash value. It seems it would always be null. @nastra @ChunxuTang was there any reason it was done this way? I think we should remove it from IcebergSessionProperties.

Reply:
@yingsu00 I think it's probably ok to remove the `nessie_reference_hash` property as I'm unaware of any usage of it. 
@nastra Do you happen to have any insights? Is it ok to remove it?

Reply:
Hi, `nessie_reference_hash` is optionally needed to fetch the state of table from particular time (mapping to the hash). Without hash, we always fetch the latest state. Old state can be useful for time travel queries. 

But as pointed out above, If it is not exposed to the user or integrated fully. We can remove for time being and add it later when we fully want to support it. 

But as @nastra suggested better to keep the current PR scope as small as possible.

Reply:
@ajantha-bhat thanks for your reply. But according to https://projectnessie.org/tools/client_config/ this ref hash is the hash of the ref name. 


nessie.ref | Mandatory | Name of the Nessie reference, usually main.
-- | -- | --
nessie.ref.hash | Optional | Hash on nessie.ref, usually not specified.


What is the ref name used for? If ref name is already part of the cache key, do we still need the ref hash?

Reply:
Nessie supports catalog level branching and tagging capabilities. ref name is (branch or tag name). 
If the hash is not specified, we use the latest state of the table from that branch or tag. 

catalog level hash configuration is mainly used for time travel queries. Once, we support that we need this config exposed. 
If it is not really causing much problem, we can even keep this code to avoid adding it back again in the future.